src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ReadOnlyTensorSpan.cs

```diff
@@ -520,31 +520,38 @@ public void CopyTo(scoped TensorSpan<T> destination)
             // Using "if (!TryCopyTo(...))" results in two branches: one for the length
             // check, and one for the result of TryCopyTo. Since these checks are equivalent,
             // we can optimize by performing the check once ourselves then calling Memmove directly.
-            if (_shape.FlattenedLength <= destination.FlattenedLength)
+            if (TensorHelpers.IsBroadcastableTo(Lengths, destination.Lengths))
             {
                 scoped Span<nint> curIndexes;
                 nint[]? curIndexesArray;
+
                 if (Rank > TensorShape.MaxInlineRank)
                 {
-                    curIndexesArray = ArrayPool<nint>.Shared.Rent(Rank);
+                    curIndexesArray = ArrayPool<nint>.Shared.Rent(destination.Rank);
                     curIndexes = curIndexesArray;
-                    curIndexes = curIndexes.Slice(0, Rank);
+                    curIndexes = curIndexes.Slice(0, destination.Rank);
```

Comment:
Other PRs have switched to the slice and clear pattern, should this be doing the same?

Reply:
Yes. Fixed. This change is going to cause a conflict with the other branch as well.